distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Enso_Types.enso

```diff
@@ -35,10 +35,14 @@ most_specific_value_type value use_smallest=False =
                 True -> Value_Type.Decimal precision=Nothing scale=0
         text : Text     ->
             length = text.length
-            # Not using Char size=0 for empty strings, because that would be an invalid value.
-            case use_smallest && length > 0 of
-                True  -> Value_Type.Char size=text.length variable_length=False
+            case use_smallest of
                 False -> Value_Type.Char size=Nothing variable_length=True
+                True -> 
+                    case length > 0 of
+                        True  -> Value_Type.Char size=length variable_length=False
+                        # Not using Char size=0 for empty strings, because that would be an invalid value.
+                        False -> Value_Type.Char size=1 variable_length=True
```

Comment:
Interesting take.

It was not immediately obvious to me why it would have a max size, but I guess it is to avoid 'enlarging' other types when they are merged with the type for `""`. Nice.

Reply:
I was worried it may not work if we do ```
c = Column.from_vector ["a", Nothing, " "] . cast (Value_Type.Char size=1 variable_length=False)
r = c.fill_missing ""
```

But I think it will work correctly.
Still, I'd be happy to see this edge case tested:
```
r.value_type . should_equal (Value_Type.Char size=1 variable_length=True)
```

Reply:
Added this test

Reply:
Looks perfect, thanks!